ToDoApp/main.py

```diff
@@ -9,5 +16,56 @@ async def root():
     return {"message": "Hello World"}
 
 
+@app.post("/user/register/", tags=["users"], response_model=str)
+async def register(*, user: schemas.UserCreate = Body(default=None), db: Session = Depends(get_DB),
+                   background_tasks: BackgroundTasks):
+    """
+    This function adds the user in DB and sends a verification email
+    :param user:
+    :param db:
+    :param background_tasks:
+    :return:
+    """
+    db_user = crud.get_user_by_email(db, user_email=user.email)
+    if db_user:
+        raise HTTPException(status_code=400, detail="Email already registered")
+    hashed_password = utils.hash_password(user.password)
+    user_db = schemas.UserCreate(email=user.email, name=user.name, password=hashed_password)
+    stored_user = crud.create_user(db=db, user=user_db)
+    token = utils.encode_token(stored_user.email)
+    # check if token already exists
+    db_token = crud.get_token(db, user_id=stored_user.id)
+    if not db_token:
+        crud.create_token(db, token=token, user_id=stored_user.id)
+        encrypted_token = utils.encrypt_token(token.access_token)
+        token = encrypted_token.decode("utf-8")
+        verification_link = f"http://localhost:8000/user/confirmation/{token}"
+        background_tasks.add_task(auth_email.send_verification_email, token=verification_link, user_email=stored_user.email)
+        return verification_link
```

Comment:
what happens in case db_token is set to some value? we won't get response

Reply:
This "if" was not needed here as we are first checking if user exists and only then we are creating and adding token in Token table.